app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  # Validations for the User model
+  validates :name, presence: true
+  validates :posts_count, numericality: { greater_than_or_equal_to: 0 }
```

Comment:
- [ ] [OPTIONAL] Please also add validation for posts counter to be only integer. See example.

```rb
 validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```